packages/components/src/components/primitives/form/DropdownField/DropdownSingleSelect/DropdownSingleSelect.tsx

```diff
@@ -39,12 +40,47 @@ export const DropdownSingleSelect = forwardRef<HTMLElement, Props>(
     }: Props,
     ref
   ) => {
+    const defaultValue = initialSelectedOption && initialSelectedOption.value;
+    const [itemSelected, setItemSelected] = useState(defaultValue || null);
+    const {
+      selectClasses,
+      buttonClasses,
+      selectTextClasses,
+      selectFocusClasses,
+      selectDisabledClasses,
+      selectErrorClasses,
+      selectListClasses,
+      itemListClasses,
+      icon,
+    } = getDropdownSingleSelectClasses(disabled, invalid, itemSelected);
+
+    if (isMobile) {
```

Comment:
please extract all the logic in this component to another file and in here just return a component or the other

Reply:
Done!